### PR TITLE
Issue 2549/autocomplete group and gpus support

### PR DESCRIPTION
**Pull request by rchenzheng**
**- What I did**
Confirm the issues presented in https://github.com/docker/cli/issues/2549 and https://github.com/docker/cli/issues/2549#issuecomment-640055524

Fix Auto complete options available:
- Add support for --gpus to run/create container in bash and zsh
- Remove --group from run and update container as it's not a valid flag in zsh
- Add --group-add --group-rm for create and update update service in bash

**- How I did it**
Update the auto complete files

**- How to verify it**
For zsh
```
/usr/local/bin/zsh

source contrib/completion/zsh/_docker
docker container run --g[TAB][TAB]
--gpu and --group-add
docker service create --g[TAB][TAB]
--group
docker service update --g[TAB][TAB]
--group-add and --group-rm

docker container --help
docker service --help
```

For bash
```
/bin/bash
 source contrib/completion/bash/docker

docker container run --g[TAB][TAB]
--gpus and --group-add
docker service create --g[TAB][TAB]
--generic-resource and --group
docker service update --g[TAB][TAB]
--generic-resource-add  --generic-resource-rm   --group-add             --group-rm 

docker container --help
docker service --help
```

I also went through the library option and tested each command and confirmed that those options were valid for docker-cli aka no 'unknown flag:`

**- Description for the changelog**

Fix Auto complete options available:
- Add support for --gpus to run/create container in bash and zsh
- Remove --group from run and update container as it's not a valid flag in zsh
- Add --group-add --group-rm for create and update update service in bash

**- A picture of a cute animal (not mandatory but encouraged)**

https://i.pinimg.com/originals/a7/da/a1/a7daa190107fa107413036bdace84884.jpg

### CODE REVIEW

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -3842,7 +3843,7 @@ _docker_service_update_and_create() {
 			__docker_nospace
 			return
 			;;
-		--group|--group-add|--group-rm)
+		--group-add|--group-rm)
```
Comment (albers) [inline contrib/completion/bash/docker:3846]: With this change, `docker service create --group <tab>` no longer completes groups.

Comment (rchenzheng) [reply contrib/completion/bash/docker:3846]: It still does in my case, can you confirm? 

```
source contrib/completion/bash/docker
source ~/.bash_profile

docker service create --g
bash-3.2$ docker service create --g
--generic-resource  --group             
bash-3.2$ docker service create --g
--generic-resource  --group             
bash-3.2$ docker service create --g
```

Comment (albers) [reply contrib/completion/bash/docker:3846]: @rchenzheng
> With this change, `docker service create --group <tab>` no longer completes groups.

The issue is about completing the arguments of the `--group` option.

The code above is only called when the preceding argument is either `--group`, `--group-add` or `--group-rm`. In the context of `service create`, `--group` is still valid and needs support for completing its arguments.

Comment (rchenzheng) [reply contrib/completion/bash/docker:3846]: Got it! Sorry, it's been a busy day

**Comment by rchenzheng** [comment]
@albers Looks like we need to rerun the job

**Comment by thaJeztah** [comment]
Could you squash your commits? Let me know if you need a hand/instructions on doing so

**Comment by rchenzheng** [comment]
> Could you squash your commits? Let me know if you need a hand/instructions on doing so

I definitely can, thought it was automated at the end

**Comment by thaJeztah** [comment]
We could use GitHub's "squash and merge" feature, but prefer to keep contributor's commits as they were authored by the contributor. GitHub's feature also removes GPG signing (which is used by some people), which we try to keep intact if possible.

**Pull request merged**